opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/jdbc/JdbcTracingTest.java

```diff
@@ -0,0 +1,105 @@
+package io.opentracing.contrib.spring.cloud.jdbc;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.contrib.jdbc.TracingConnection;
+import io.opentracing.contrib.spring.cloud.MockTracingConfiguration;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.GlobalTracer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import javax.sql.DataSource;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+@SpringBootTest(classes = {MockTracingConfiguration.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+public class JdbcTracingTest {
+
+    @Autowired
+    MockTracer tracer;
+
+    @Autowired
+    DataSource dataSource;
+
+    @Autowired
+    JdbcTemplate jdbcTemplate;
+
+    @Before
+    public void before() {
+        tracer.reset();
+    }
+
+    /**
+     * Make sure we get a data source that returns a tracing connection. Without our Aspect, this would return a regular
+     * pooled connection for an embedded database (as we don't configure a database anywhere).
+     */
+    @Test
+    public void dataSourceIsInstrumented() throws SQLException {
+        assertTrue(dataSource.getConnection() instanceof TracingConnection);
+    }
+
+    /**
+     * Make sure we get one span once we execute a database call.
+     */
+    @Test
+    public void spanIsCreatedForPreparedStatement() throws SQLException {
+        PreparedStatement pstmt = dataSource.getConnection().prepareStatement("select 1");
+        assertTrue(pstmt.execute());
+        assertEquals(1, tracer.finishedSpans().size());
+    }
+
+    /**
+     * Make sure that a span is created when an active span exists joins the active
+     * @throws SQLException
+     */
+    @Test
+    public void spanJoinsActiveSpan() throws SQLException {
+        try (ActiveSpan ignored = GlobalTracer.get().buildSpan("parent").startActive()) {
```

Comment:
nit use tracer directly. You have it autowired here.

Reply:
I actually did it on purpose: I want to make sure that a span created by the tracer at the `GlobalTracer` can be a parent to a span created by this tracer. This is perhaps a test for the `ot-jdbc`, not for the Spring OT JDBC integration, but still :)

Reply:
We test this here https://github.com/opentracing-contrib/java-spring-web/blob/master/opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/TracerAutoConfigurationTest.java#L35

Reply:
Although it's weird that `java-jdbc` depends only on `GlobalTracer` and it's not possible to specify the tracer explicitly